### ISSUE TITLE
V2 --> V3 migration

### ISSUE THREAD

**Issue by jshorland** [ushahidi/platform#1410]
We need to identify everything in V2 we're missing in V3. So far I have:
- category structure (category, sub categories vs surveys, categories)
- comments
- news links

Is there anything else I'm missing here @aodour and @zhalloran?

We'll also need to address how the migration will happen. Automated tool? What potential problems do we need to flag out of the gate?

Aha! Link: https://ushahiditeam.aha.io/features/PROD-664

**Comment by jshorland** [comment]
@kamaulynder would love your input on this as well if you have time.

**Comment by rjmackay** [comment]
There are a few old issue list here: https://github.com/ushahidi/platform/issues?q=is%3Aopen+is%3Aissue+milestone%3A%22V2+to+V3+Migration%22 
They include
- Adding custom pages to deployments ie. about us, etc
- Contact Us page (for contacting the deployer)

As I see it there are a few likely stages in building migrations:

1. Building missing features / or deciding to EOL a feature
2. Building a prototype importer
   * this might be something only dev friendly, but usable for one off requests.
3. Community usable importer
   * has a UI, should be usable by most people. 
   * This might require access to a V2 DB, or just a V2 API. Need to scope that.
4. Hosted importer for Crowdmap -> .io, self hosted v2 -> .io, etc.
   * Depending on needs we might not need to do this (ie. if demand is low and we can do manual migrations on request)

1 & 2 can actually happen in parallel.

**Comment by aoduor** [comment]
I'm going to add notes i'd made a while back on this, and the we can work out what's feasible/isn't. Feel free to reject anything that seems out of scope(and apologies in advance for the deluge :D).

**General Settings**
- Site copyright statement - help deployers determine if others have rights to reuse their content/data
- Cache pages/cache timeline? - will greatly 
- Contact page ( this is huge)
- Set administrator with ability to manually approve/unapprove user sign ups on their deployment?

**Map settings**
- Additional map base layers

**API Settings**
- This is missing altogether and would be great for limiting access to the API/Banning access

**Data sources**
- Support for frontlineSMS 
- Ability for user to manage frequency of data coming in(the scheduler)

**Surveys**
- V2 had the option to determine submission and viewing of responses by user group for each field type. Example - if i want to collect user contact info, but don’t want to have that displayed publicly
- Admin should have the option to activate/deactivate post types, or set them to be visible to a select group of people
- Posts from email/SMS
-- Users need to be able to respond to SMS from within the platform
-- Users should also be able to see conversation threads from a single source

** Categories**
- Blue sky, but we should also add the ability for users to curate translations for their categories like we did in v2
- Admin should also be able to set what categories are visible/hidden to certain user roles

**User Management**
- The admin should either
-- Be able to set which custom role he/she would like assigned to any new user signing up on their deployment OR
-- * The "User" role should not be editable/ deletable(is that even a word :P). 
- In v2 we had the "Superadmin" and "Member" roles as defaults that could not be modified/deleted.
- Should we also let the admin determine whether he’d like users to sign themselves up on a deployment/approve user sign ups?

**Alerts**
Right now we only have the option to add notifications for registered users on saved searches and collections. We need to open this up.

**Custom Pages**
This one is missing altogether. It’s useful for users to have the ability to add custom pages e.g about us etc.

**General questions**
- Language toggle button on homescreen as opposed to leaving this to the admin only
- Comments? Spam control?

**Comment by aoduor** [comment]
@jshorland @rjmackay perhaps this would be better as a milestone with individual issues tagged to it?

**Comment by Erioldoesdesign** [comment]
Adding my post-it notes sketches from my sync with @tuxpiper re. v2->v3 here because I don't think they apply to some of the more specific tickets just yet...

![IMG_20191031_093028](https://user-images.githubusercontent.com/11681324/68409172-89f0fb00-017e-11ea-9a6f-0f7f9a88cf56.jpg)
![IMG_20191031_093037](https://user-images.githubusercontent.com/11681324/68409178-8e1d1880-017e-11ea-9c21-7cddc9a17445.jpg)
![IMG_20191031_093038](https://user-images.githubusercontent.com/11681324/68409188-91180900-017e-11ea-9841-21e96382f554.jpg)
![IMG_20191031_093039](https://user-images.githubusercontent.com/11681324/68409194-92e1cc80-017e-11ea-9895-76e5bc95c678.jpg)
![IMG_20191031_093042](https://user-images.githubusercontent.com/11681324/68409212-99704400-017e-11ea-9992-35ebbef837ea.jpg)
![IMG_20191031_093046](https://user-images.githubusercontent.com/11681324/68409231-a0975200-017e-11ea-8d0b-8eef4852d43e.jpg)
![IMG_20191031_093044](https://user-images.githubusercontent.com/11681324/68409236-a42ad900-017e-11ea-9d4f-d7a32053e4c6.jpg)
